recipes/lfortran/build.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -ex
+
+printenv
+
+if [[ ${MACOSX_DEPLOYMENT_TARGET} != "" ]]; then
+    # macOS
+    python -m pip install . -vv \
+      --install-option="bdist_wheel" \
+      --install-option="--plat-name" \
+      --install-option="macosx-10.9-x86_64"
```

Comment:
Use `${MACOSX_DEPLOYMENT_TARGET}` instead of 10.9

Reply:
Good point. Do you understand what the issue is here? There might be an easier fix in the CMakeLists.txt than fiddling with things at the pip level. It seems there is a failure later on that perhaps could be avoided if CMake can use the proper `MACOSX_DEPLOYMENT_TARGET` on its own.

Reply:
skbuild will send `-DCMAKE_OSX_DEPLOYMENT_TARGET=10.6` to cmake overriding the environment variable that CMake usually uses.

Reply:
skbuild also ignores `CFLAGS, LDFLAGS` env variables as well, which is annoying.

Reply:
It almost seems it's a bug in skbuild that we are trying to patch here. Since skbuild is sending the wrong `CMAKE_OSX_DEPLOYMENT_TARGET`.